src/state/lists/wrappedTokenInfo.ts

```diff
@@ -0,0 +1,63 @@
+import { Tags, TokenInfo, TokenList } from '@uniswap/token-lists'
+
+import { isAddress } from '../../utils'
+import { Token } from '@dynamic-amm/sdk'
+
+type TagDetails = Tags[keyof Tags]
+interface TagInfo extends TagDetails {
+  id: string
+}
+/**
+ * Token instances created from token info on a token list.
+ */
+export class WrappedTokenInfo extends Token {
```

Comment:
Just FYI, https://github.com/Uniswap/sdk-core/blob/main/src/entities/currency.ts#L4

as I checked before, Uniswap sdk had refactored `Currency = NativeCurrency | Token`

while in our sdk, `Token extends Currency`.
Just be careful here, because if it affects something else, we will have lots of things to refactor.

Reply:
yeah, I saw it. They add some field to Currency like isToken, isNative.